### PR TITLE
Add pt-fill support for InputGroups

### LINKED ISSUE #2339: Support pt-fill on InputGroups
InputGroups do not support pt-fill outside of ControlGroups, so I cant replace text inputs with InputGroups in responsive layouts. It should be pretty straightforward to support it, so I can create a PR if no team members want to tackle it.

### PR DESCRIPTION
**Pull request by qcz**
#### Checklist
- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [x] Update documentation

#### Changes proposed in this pull request:
* Support pt-fill on InputGroups

#### Reviewers should focus on:
* Does this change have any effects on other Blueprint components?

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @qcz! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by qcz** [comment]
<h3>Add pt-fill support for InputGroups</h3>

Preview: <strong><a href='https://13-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html' target='_blank'>documentation</a> | <a href='https://13-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/landing-app/dist/index.html' target='_blank'>landing</a> | <a href='https://13-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/table-dev-app/dist/index.html' target='_blank'>table</a></strong>

File: packages/core/src/components/forms/_input-group.scss [hunk 1]
```diff
@@ -27,6 +27,7 @@ Markup:
 .pt-round - Rounded caps. Button will also be rounded.
 .pt-large - Large group. Children will adjust size accordingly.
 .pt-intent-primary - Primary intent. (All 4 intents are supported.)
+.pt-fill - Take up full width of parent element
```
Comment (giladgray) [inline packages/core/src/components/forms/_input-group.scss]: nit: trailing period please (like the lines above)

Comment (qcz) [reply packages/core/src/components/forms/_input-group.scss]: Copied from `_inputs.scss` like this. Will fix.

**Comment by qcz** [comment]
It seems that an additional `flex: 1 1 auto` will fix all the cases I tried.
However simple `.pt-input`s are also affected by this problem, so it is not entirely new:
firefox_2018-04-04_20-46-47
Should I fix that in this PR too?

**Comment by giladgray** [comment]
@qcz we can't merge this until any outstanding issues are resolved so yes please keep pushing.

**Comment by qcz** [comment]
<h3>Add proper flex support for .pt-input and .pt-input-group with .pt-fill</h3>

Preview: <strong><a href='https://19-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html' target='_blank'>documentation</a> | <a href='https://19-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/landing-app/dist/index.html' target='_blank'>landing</a> | <a href='https://19-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/table-dev-app/dist/index.html' target='_blank'>table</a></strong>

**Comment by qcz** [comment]
<h3>Fix lint errors</h3>

Preview: <strong><a href='https://25-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html' target='_blank'>documentation</a> | <a href='https://25-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/landing-app/dist/index.html' target='_blank'>landing</a> | <a href='https://25-127462892-gh.circle-artifacts.com/0/home/circleci/project/packages/table-dev-app/dist/index.html' target='_blank'>table</a></strong>

**Pull request merged**